src/openssl.cr

```diff
@@ -1,5 +1,54 @@
 require "./openssl/lib_ssl"
 
+# ### Example
+# ## Server side
+# for the below "server" example to work, a key pair should be created, using openssl it can be done like that
+# - Generate keys to /tmp/
+# - openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/private.key -out /tmp/certificate.crt
+# ```crystal
+# require "socket"
+# require "openssl"
+#
+# def server
+#   socket = TCPServer.new(5555) # Bind new TCPSocket to port 5555
+#   context = OpenSSL::SSL::Context::Server.new
+#   # Define which ciphers to use with OpenSSL
+#   # recommended ciphers can be taken from
+#   # - https://www.owasp.org/index.php/Transport_Layer_Protection_Cheat_Sheet#Rule_-_Only_Support_Strong_Cryptographic_Ciphers
+#   # - https://cipherli.st/
+#   # - Full list is available at: https://wiki.openssl.org/index.php/Manual:Ciphers(1)#CIPHER_STRINGS
+#   context.ciphers = "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH"
```

Comment:
I'm not sure about adding a valid value here, people should be forced to actually follow the links given rather than copy pasting whatever is here. Also it should be made clear that we do provider a sane enough default.